mesonbuild/modules/pkgconfig.py

```diff
@@ -136,6 +138,26 @@ def generate(self, state, args, kwargs):
         pub_reqs = mesonlib.stringlistify(kwargs.get('requires', []))
         priv_reqs = mesonlib.stringlistify(kwargs.get('requires_private', []))
         conflicts = mesonlib.stringlistify(kwargs.get('conflicts', []))
+
+        def parse_variable_list(stringlist):
+            reserved = ['prefix', 'libdir', 'includedir']
+            variables = []
+            for var in stringlist:
+                try:
+                    k, v = var.split('=')
+                    if len(k) == 0 or len(v) == 0:
```

Comment:
You don't want to use `len` here: `if not k or not v` will do the same thing, but is about 10x faster, as well as less code.